index.js

```diff
@@ -44,6 +45,44 @@ Counter.prototype.add = function(id, options){
 }
 
 /**
+ *    
+ * Note: Performance and memory usage could be greatly improved by:
+ * 1. removing the need for a newly allocated 'options' object everytime 'add' method is called
+ * 2. replacing 'options' with primitive types (Number, Boolean)
+ * This would lower the load on the GC during its garbage collection cycle, reduce heap memory footprint, and lower execution time
+ * 
+ * Adds an element to the counter, or increases
+ * its quantity if it was already in the counter.
+ * 
+ *
+ *
+ * Adding might fail if there is a reference that
+ * hold a lower count for the given ID. You can
+ * specify parameter force: true in the options to override
+ * this behaviour.
+ *
+ * @param  {String} id
+ * @param  {Number} qty  
+ * @param  {Boolean} force 
+ * @return {Boolean}         Whether adding has been succesfull or not
+ */
+Counter.prototype.addStrict = function(id, qty, force){
```

Comment:
Hello Alessandro,

If you can try to run the benchmark.js (counting 100000000 times) you can notice some performance data as:

**addStrict** function:

**Before execution**: { rss: 26148864, heapTotal: 11327232, **heapUsed**: 6069120 }
**After excution**: { rss: 26689536, heapTotal: 12359168, **heapUsed**: 6423784 }

benchmark execution time -> **251ms**

Summary:

usedHeap: 354,664
executime time: 251ms

----------------

**add** function:

**Before execution**: { rss: 26689536, heapTotal: 12359168, **heapUsed**: 6486344 }
**After excution**: { rss: 31285248, heapTotal: 16486912, **heapUsed**: 7840824 }

benchmark execution time -> **1156ms**

Summary:

usedHeap: 1,354,480
executime time: 1156ms

You can notice the memory foot print (usedHeap) and the execution time is a lot less while using the 'addStrict' method. The purpose of the change is to **avoid** 'options' object allocation everytime we call the *add* function, so instead of passing an object, you could replace it with primitive types (Number, Boolean) arguments.

Please let me know if this makes any sense.